### PR TITLE
redhat_subscription: Add server_prefix and server_port as supported arguments

### PR DESCRIPTION
**Pull request by unnecessary-username**
##### SUMMARY

This is Tong He from Red Hat MPQE Entitlement QE Team.

server.prefix and server.port are two arguments in 'subscription-manager config' but now not supported in this module. As I see more than one requests mentioning these, I think it is valuable to add these two as supported arguments.

This PR is in response to and implements the following feature requests:
Fixes https://github.com/ansible-collections/community.general/issues/1311
https://bugzilla.redhat.com/show_bug.cgi?id=1766342

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

redhat_subscription

##### ADDITIONAL INFORMATION

An example of the contents in subscription-manager config:
```paste below
[server]
   hostname = [subscription.rhsm.redhat.com]
   insecure = [0]
   no_proxy = []
   port = [443]
   prefix = [/subscription]
   proxy_hostname = []
   proxy_password = []
   proxy_port = []
   proxy_scheme = [http]
   proxy_user = []
   server_timeout = [180]
   ssl_verify_depth = [3]

[rhsm]
   auto_enable_yum_plugins = [1]
   baseurl = [https://cdn.redhat.com]
   ca_cert_dir = [/etc/rhsm/ca/]
   consumercertdir = [/etc/pki/consumer]
   entitlementcertdir = [/etc/pki/entitlement]
   full_refresh_on_yum = [0]
   inotify = [1]
   manage_repos = [1]
   package_profile_on_trans = [0]
   pluginconfdir = [/etc/rhsm/pluginconf.d]
   plugindir = [/usr/share/rhsm-plugins]
   productcertdir = [/etc/pki/product]
   repo_ca_cert = /etc/rhsm/ca/redhat-entitlement-authority.pem
   repomd_gpg_url = []
   report_package_profile = [1]

[rhsmcertd]
   autoattachinterval = [1440]
   certcheckinterval = [240]
   disable = [0]
   splay = [1]

[logging]
   default_log_level = [INFO]

[] - Default value in use


```

### CODE REVIEW

File: plugins/modules/packaging/os/redhat_subscription.py [hunk 1]
```diff
@@ -46,6 +46,16 @@
         description:
             - Enable or disable https server certificate verification when connecting to C(server_hostname)
         type: str
+    server_prefix:
+        description:
+            - Specify the prefix when registering to the Red Hat Subscription Management or Sat6 server
```
Comment (felixfontein) [inline plugins/modules/packaging/os/redhat_subscription.py:51]:
```suggestion
            - Specify the prefix when registering to the Red Hat Subscription Management or Sat6 server.
```
Same below.

Comment (unnecessary-username) [reply plugins/modules/packaging/os/redhat_subscription.py]: Resolved in the latest commit, please have a check.

File: changelogs/fragments/2779_redhat_subscription-add_server_prefix_and_server_port.yml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - redhat_subscription - add server_prefix and server_port as supported arguments (https://github.com/ansible-collections/community.general/pull/2779).
```
Comment (aminvakil) [inline changelogs/fragments/2779_redhat_subscription-add_server_prefix_and_server_port.yml:2]:
```suggestion
  - redhat_subscription - add ``server_prefix`` and ``server_port`` paramaters (https://github.com/ansible-collections/community.general/pull/2779).
```

**Review by aminvakil — COMMENTED** [review]
Thanks for adding this!

**Review by felixfontein — APPROVED** [review]

File: changelogs/fragments/2779_redhat_subscription-add_server_prefix_and_server_port.yml [hunk 3]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - redhat_subscription - add ``server_prefix`` and ``server_port`` paramaters (https://github.com/ansible-collections/community.general/pull/2779).
```
Comment (aminvakil) [inline changelogs/fragments/2779_redhat_subscription-add_server_prefix_and_server_port.yml:2]: Sorry, I never learn the spell of this damned word!
```suggestion
  - redhat_subscription - add ``server_prefix`` and ``server_port`` parameters (https://github.com/ansible-collections/community.general/pull/2779).
```

**Review by aminvakil — COMMENTED** [review]
I guess it would be nice to have an example with these new parameters too, but it's fine as it is now as well, please fix the spelling problem and add some examples if you also think it's good, otherwise I'll merge this after the typo fix.

File: changelogs/fragments/2779_redhat_subscription-add_server_prefix_and_server_port.yml (hunk 3 above)
Comment (felixfontein) [reply changelogs/fragments/2779_redhat_subscription-add_server_prefix_and_server_port.yml:2]: Ooops, I completely missed that...

Comment (unnecessary-username) [reply changelogs/fragments/2779_redhat_subscription-add_server_prefix_and_server_port.yml]: These two parameters would not be used frequently so I did not plan to add examples on them. Thank you. (I also missed to spot the 'paramaters' there)

**Review by aminvakil — APPROVED** [review]

**Comment by aminvakil** [comment]
@unnecessary-username Thanks for adding these new parameters!
@felixfontein @russoz Thanks for reviewing!

**Pull request merged**